src/GitHub.App/Factories/ViewViewModelFactory.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Windows;
+using GitHub.Exports;
+using GitHub.Services;
+using GitHub.ViewModels;
+
+namespace GitHub.Factories
+{
+    /// <summary>
+    /// Factory for creating views and view models.
+    /// </summary>
+    [Export(typeof(IViewViewModelFactory))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class ViewViewModelFactory : IViewViewModelFactory
+    {
+        readonly IGitHubServiceProvider serviceProvider;
+
+        [ImportingConstructor]
+        public ViewViewModelFactory(
+            IGitHubServiceProvider serviceProvider,
+            ICompositionService cc)
+        {
+            this.serviceProvider = serviceProvider;
+            cc.SatisfyImportsOnce(this);
+        }
+
+        [ImportMany(AllowRecomposition = true)]
+        IEnumerable<ExportFactory<FrameworkElement, IViewModelMetadata>> Views { get; set; }
+
+        /// <inheritdoc/>
+        public TViewModel CreateViewModel<TViewModel>() where TViewModel : IViewModel
+        {
+            return serviceProvider.ExportProvider.GetExport<TViewModel>().Value;
```

Comment:
We could do the same as with `CreateView` and avoid any dependency on `IGitHubServiceProvider`.

Reply:
Yep! Lets try that in a separate PR too.